packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -497,72 +480,56 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
-            intermediateAncillaryData,
-            "depositId",
-            uint256(_depositData.depositId)
-        );
-        intermediateAncillaryData = AncillaryData.appendKeyValueAddress(
-            intermediateAncillaryData,
-            "l1Recipient",
-            _depositData.l1Recipient
-        );
-        intermediateAncillaryData = AncillaryData.appendKeyValueAddress(
-            intermediateAncillaryData,
-            "l2Sender",
-            _depositData.l2Sender
-        );
-        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
-            intermediateAncillaryData,
-            "amount",
-            _depositData.amount
-        );
-        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
-            intermediateAncillaryData,
-            "slowRelayFeePct",
-            uint256(_depositData.slowRelayFeePct)
-        );
-        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
-            intermediateAncillaryData,
-            "instantRelayFeePct",
-            uint256(_depositData.instantRelayFeePct)
-        );
-        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
-            intermediateAncillaryData,
-            "quoteTimestamp",
-            uint256(_depositData.quoteTimestamp)
-        );
+        return
+            AncillaryData.getKeyValueBytes32(
+                "hash",
+                keccak256(
+                    abi.encode(
+                        _depositData.chainId,
+                        _depositData.depositId,
+                        _depositData.l1Recipient,
+                        _depositData.l2Sender,
+                        _depositData.amount,
+                        _depositData.slowRelayFeePct,
+                        _depositData.instantRelayFeePct,
+                        _depositData.quoteTimestamp,
+                        _relayData.relayId,
+                        _relayData.realizedLpFeePct,
+                        address(l1Token)
+                    )
+                )
+            );
+    }
 
-        // Add relay data.
-        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
-            intermediateAncillaryData,
-            "relayId",
-            uint256(_relayData.relayId)
-        );
-        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
-            intermediateAncillaryData,
-            "realizedLpFeePct",
-            uint256(_relayData.realizedLpFeePct)
-        );
+    // Note: this method is identical to the one above, but it allows storage to be passed in, which saves some gas (3-4k)
+    // when called internally due to solidity not needing to copy the entire data structure and just lazily read data
+    // when requested.
+    function _getRelayAncillaryData(bytes32 relayHash) private view returns (bytes memory) {
+        // TODO: Consider adding BridgePool address to the ancillary data packet.
```

Comment:
I think this todo is not necessary and can be removed. The bridge pool address is always linked to the event that it emits

Reply:
Sounds good -- will remove!